internal/cmd/logout.go

```diff
@@ -2,31 +2,76 @@ package cmd
 
 import (
 	"errors"
+	"fmt"
+	"os"
 
 	"github.com/spf13/cobra"
 
 	"github.com/infrahq/infra/internal/logging"
 )
 
 func newLogoutCmd() *cobra.Command {
-	var purge bool
+	var (
+		clear  bool
+		server string
+		all    bool
+	)
 
 	cmd := &cobra.Command{
-		Use:     "logout",
-		Short:   "Log out of Infra",
-		Example: "$ infra logout",
-		Group:   "Core commands:",
+		Use:   "logout [SERVER]",
+		Short: "Log out of Infra",
+		Example: `# Log out of current server
+$ infra logout
+		
+# Log out of a specific server
+$ infra logout infraexampleserver.com
+		
+# Logout of all servers
+$ infra logout --all 
+		
+# Log out of current server and clear from list 
+$ infra logout --clear
+		
+# Log out of a specific server and clear from list
+$ infra logout infraexampleserver.com --clear 
+		
+# Logout and clear list of all servers 
+$ infra logout --all --clear`,
+		Args:  cobra.MaximumNArgs(1),
+		Group: "Core commands:",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return logout(purge)
+			if len(args) == 1 {
+				if all {
+					fmt.Fprintf(os.Stderr, "  Server is already specified. Ignoring flag [--all] and logging out of server %s.\n", args[0])
```

Comment:
ignoring user input is usually not a good thing. it's possible they're adding `--all` to `logout SERVER` expecting it to logout of all servers